lib/tooling/base-tool.js

```diff
@@ -24,19 +24,31 @@
 
 import path from 'path';
 
+import PromClient from 'prom-client';
 import _ from 'underscore';
 
 import * as exec from '../exec';
 import {logger} from '../logger';
 import * as utils from '../utils';
 
+const countersPerTool = {};
+
 export class BaseTool {
     constructor(toolInfo, env) {
         this.tool = toolInfo;
         this.env = env;
         this.tool.exclude = this.tool.exclude ? this.tool.exclude.split(':') : [];
         this.addOptionsToToolArgs = true;
         this.parseOutput = utils.parseOutput;
+
+        const toolName = this.tool.name;
+        if (toolName && !(toolName in countersPerTool)) {
+            countersPerTool[toolName] = new PromClient.Counter({
+                name: `${toolName.replaceAll(/\W/g, '_').toLowerCase()}_compilation_total`,
+                help: `Number of compilations with ${toolName} active`,
+                labelNames: ['language', 'compiler'],
```

Comment:
I'm not sure the `compiler` is useful here? Tools run independently of the compiler (I think?)